core/src/shred_fetch_stage.rs

```diff
@@ -434,170 +425,3 @@ fn check_feature_activation(
         }
     }
 }
-
-#[cfg(test)]
-mod tests {
-    use {
-        super::*,
-        solana_ledger::{
-            blockstore::MAX_DATA_SHREDS_PER_SLOT,
-            shred::{ReedSolomonCache, Shred, ShredFlags},
-        },
-        solana_sdk::packet::Packet,
-    };
-
-    #[test]
-    fn test_data_code_same_index() {
-        solana_logger::setup();
-        let mut packet = Packet::default();
-        let mut stats = ShredFetchStats::default();
-
-        let slot = 2;
-        let shred_version = 45189;
-        let shred = Shred::new_from_data(
-            slot,
-            3,   // shred index
-            1,   // parent offset
-            &[], // data
-            ShredFlags::LAST_SHRED_IN_SLOT,
-            0, // reference_tick
-            shred_version,
-            3, // fec_set_index
-        );
-        shred.copy_to_packet(&mut packet);
-
-        let last_root = 0;
-        let last_slot = 100;
-        let slots_per_epoch = 10;
-        let max_slot = last_slot + 2 * slots_per_epoch;
-        assert!(!should_discard_shred(
-            &packet,
-            last_root,
-            max_slot,
-            shred_version,
-            |_| false, // should_drop_legacy_shreds
-            |_| true,  // enable_chained_merkle_shreds
-            &mut stats,
-        ));
-        let coding = solana_ledger::shred::Shredder::generate_coding_shreds(
-            &[shred],
-            3, // next_code_index
-            &ReedSolomonCache::default(),
-        );
-        coding[0].copy_to_packet(&mut packet);
-        assert!(!should_discard_shred(
-            &packet,
-            last_root,
-            max_slot,
-            shred_version,
-            |_| false, // should_drop_legacy_shreds
-            |_| true,  // enable_chained_merkle_shreds
-            &mut stats,
-        ));
-    }
-
-    #[test]
-    fn test_shred_filter() {
```

Comment:
This test is a duplicate of `ledger::shred::test_should_discard_shred`.
The test is creating legacy shreds which are now always discarded, so I removed the test.